python/lsst/ts/Dome/mock_controller.py

```diff
@@ -143,120 +211,238 @@ async def cmd_loop(self, reader, writer):
-            "status": az_motion,
-            "positionError": 0.0,
-            "positionActual": self.az_current_azimuth,
-            "positionCmd": 0.0,
-            "driveTorqueActual": [0.0, 0.0, 0.0, 0.0, 0.0],
-            "driveTorqueError": [0.0, 0.0, 0.0, 0.0, 0.0],
-            "driveTorqueCmd": [0.0, 0.0, 0.0, 0.0, 0.0],
-            "driveCurrentActual": [0.0, 0.0, 0.0, 0.0, 0.0],
-            "driveTempActual": [20.0, 20.0, 20.0, 20.0, 20.0],
-            "encoderHeadRaw": [0.0, 0.0, 0.0, 0.0, 0.0],
-            "encoderHeadCalibrated": [0.0, 0.0, 0.0, 0.0, 0.0],
-            "resolverRaw": [0.0, 0.0, 0.0, 0.0, 0.0],
-            "resolverCalibrated": [0.0, 0.0, 0.0, 0.0, 0.0],
-        }
-        return amcs_state
-
-    def determine_el_state(self):
-        el_motion = "Stopped"
-        self.log.debug(f"self.el_motion = {self.el_motion}")
-        if self.el_motion != "Stopped":
-            el_motion = (
-                self.el_motion + " to elevation " + str(self.el_motion_elevation)
-            )
-            if self.el_current_elevation < self.el_motion_elevation:
-                self.el_current_elevation = self.el_current_elevation + 5
-                if self.el_current_elevation >= self.el_motion_elevation:
-                    self.el_current_elevation = self.el_motion_elevation
-                    self.el_motion = "Stopped"
-            else:
-                self.el_current_elevation = self.el_current_elevation - 5
-                if self.el_current_elevation <= self.el_motion_elevation:
-                    self.el_current_elevation = self.el_motion_elevation
-                    self.el_motion = "Stopped"
-        apcs_state = {
-            "status": el_motion,
-            "positionError": 0.0,
-            "positionActual": self.el_current_elevation,
-            "positionCmd": 0.0,
-            "driveTorqueActual": [0.0, 0.0, 0.0, 0.0],
-            "driveTorqueError": [0.0, 0.0, 0.0, 0.0],
-            "driveTorqueCmd": [0.0, 0.0, 0.0, 0.0],
-            "driveCurrentActual": [0.0, 0.0, 0.0, 0.0],
-            "driveTempActual": [20.0, 20.0, 20.0, 20.0],
-            "resolverHeadRaw": [0.0, 0.0, 0.0, 0.0],
-            "resolverHeadCalibrated": [0.0, 0.0, 0.0, 0.0],
-            "powerAbsortion": 0.0,
-        }
-        return apcs_state
-
     async def move_az(self, **kwargs):
-        self.log.info(f"Received command 'moveAz' with arguments {kwargs}")
-        self.az_motion_azimuth = float(kwargs["azimuth"])
-        self.az_motion = "Moving"
+        """Mock moving the dome.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain the key "azimuth" with a
+            float value.
+        """
+        self.log.debug(f"Received command 'moveAz' with arguments {kwargs}")
+
+        # No conversion from radians to degrees needed since both the commands and the mock az controller
+        # use degrees.
```

Comment:
Please consider mocking the vendor's API (e.g. the TCP/IP port). The big advantage is that the only mock-specific code in the CSC is the code to start and stop the mock controller. Everything else works the same for the mock and real controller, and unit tests of the CSC exercise more of the "real" code.